src/main/java/org/digitalstack/chess/ChessBoard.java

```diff
@@ -12,10 +12,33 @@ public ChessBoard() {
     }
 
     public void add(Pawn pawn, int xCoordinate, int yCoordinate, PieceColor pieceColor) {
-        throw new UnsupportedOperationException("Need to implement ChessBoard.add()");
+       boolean isValidPosition = isLegalBoardPosition(xCoordinate, yCoordinate);
+       pawn.setChessBoard(this);
+       if(isValidPosition){
+           if(yCoordinate <= 6){
+               if(pieces[xCoordinate][yCoordinate] == null){
```

Comment:
Combine the 3 if statements by using &&, this way you will have a single if/else